External/FEXCore/Source/Interface/Core/JIT/Arm64/VectorOps.cpp

```diff
@@ -3027,6 +3027,37 @@ DEF_OP(VUABDL) {
   }
 }
 
+DEF_OP(VUABDL2) {
+  const auto Op = IROp->C<IR::IROp_VUABDL2>();
+  const auto OpSize = IROp->Size;
+
+  const auto ElementSize = Op->Header.ElementSize;
+  const auto Is256Bit = OpSize == Core::CPUState::XMM_AVX_REG_SIZE;
+
+  const auto Dst = GetVReg(Node);
+  const auto Vector1 = GetVReg(Op->Vector1.ID());
+  const auto Vector2 = GetVReg(Op->Vector2.ID());
+
+  LOGMAN_THROW_AA_FMT(ElementSize == 2 || ElementSize == 4 || ElementSize == 8, "Invalid size");
+  const auto SubRegSize =
+    ElementSize == 2 ? ARMEmitter::SubRegSize::i16Bit :
+    ElementSize == 4 ? ARMEmitter::SubRegSize::i32Bit :
+    ElementSize == 8 ? ARMEmitter::SubRegSize::i64Bit : ARMEmitter::SubRegSize::i8Bit;
+
+  if (HostSupportsSVE && Is256Bit) {
+    // To mimic the behavior of AdvSIMD UABDL, we need to get the
+    // absolute difference of the even elements (UADBLB), get the
+    // absolute difference of the odd elemenets (UABDLT), then
+    // interleave the results in both vectors together.
+
+    uabdlb(SubRegSize, VTMP1.Z(), Vector1.Z(), Vector2.Z());
+    uabdlt(SubRegSize, VTMP2.Z(), Vector1.Z(), Vector2.Z());
+    zip2(SubRegSize, Dst.Z(), VTMP1.Z(), VTMP2.Z());
```

Comment:
Oh, actually, what's the plan with regards to 512-bit SVE (or wider)? On 512-bit, zip2 would take bits 256 to 512, when you really want bits 128 to 256, right?

Reply:
Yeah... My concern is that, if the plan is to ignore the high half of the registers, zip2 won't work, because it *only* reads from the high halves of its input registers (in this case the result of uabdlt/b, based on the to-be-ignored high-halves of the input registers).

Reply:
Good catch, we'll have to revisit this once there are 512-bit SVE2 CPUs on the market. We can end up testing using the vixl simulator in the future.
This won't be the only IR operation that is broken because of this.